File: backend/routers/ocr.py
"""OCR workflow endpoints — orchestration only, no OCR logic here."""
from __future__ import annotations

from fastapi import APIRouter, Query

from backend.schemas import OcrResultList, OcrRunResult, OcrStatus
from backend.services import ocr as ocr_svc

router = APIRouter(prefix="/ocr", tags=["ocr"])


@router.post("/run", response_model=OcrRunResult)
def run(
    reprocesar_todo: bool = Query(False, description="Ignorar lo ya procesado"),
) -> OcrRunResult:
    result = ocr_svc.run_ocr(reprocesar_todo=reprocesar_todo, workers=1)
    return OcrRunResult(**result)


@router.get("/status", response_model=OcrStatus)
def status() -> OcrStatus:
    return OcrStatus(**ocr_svc.read_status())


@router.get("/results", response_model=OcrResultList)
def results() -> OcrResultList:
    items = ocr_svc.read_results()
    return OcrResultList(count=len(items), items=items)

File: automation_system/frontend/src/api/ocr.ts
import { apiFetch } from "./client";
import type { OcrResultList, OcrRunResult, OcrStatus } from "../types";

export function runOcr(reprocesarTodo = false): Promise<OcrRunResult> {
  const query = reprocesarTodo ? "?reprocesar_todo=true" : "";
  return apiFetch<OcrRunResult>(`/api/ocr/run${query}`, { method: "POST" });
}

export function fetchOcrStatus(): Promise<OcrStatus> {
  return apiFetch<OcrStatus>("/api/ocr/status");
}

export function fetchOcrResults(): Promise<OcrResultList> {
  return apiFetch<OcrResultList>("/api/ocr/results");
}